### ISSUE TITLE
`robust` is ignored when `cluster_col` is set

### ISSUE THREAD

**Issue by benslack19** [CamDavidsonPilon/lifelines#1598]
TL;DR: When using `CoxPHFitter.fit(), it doesn't matter whether a value for `robust` is specified. If there's a `cluster_col` specified, then presumably the `Huber sandwich estimator` will always be used.

I was using `cluster_col` in the `CoxPHFitter` and saw in the [docstring](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/fitters/coxph_fitter.py#L212-L214) that the sandwich estimator automatically gets used. I was aiming to match the standard errors in a test case with a `CoxTimeVarying` model by setting `robust` to the same value in the `CoxPHFitter` and `CoxTimeVarying`. (This explains my test data below.) However, I saw from issue #544 that the  `CoxTimeVarying` has not been implemented leaving me only the option to set `robust=False` in the `CoxPHFitter` model. For the test case, I can just leave `cluster_col` unspecified. **I think an error or error message should be returned in the case of `cluster_col` being set and `robust=False`.** It looks like [this conditional](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/fitters/coxph_fitter.py#L777-L778) needs to be edited.

Here's a reproducible example with my comments:

```python
import numpy.testing as npt
import pandas as pd
from lifelines import CoxPHFitter, CoxTimeVaryingFitter
from lifelines.datasets import load_stanford_heart_transplants
from lifelines.utils import to_long_format

stanford = load_stanford_heart_transplants()

# Keep only the last record for each subject, drop all covariate columns except age to simplify data
stanford_last = (
    stanford.groupby("id")
    .tail(1)
    .drop(["year", "surgery", "transplant"], axis="columns")
)

# Format the data for CPH model
stanford_last_cph_wid = stanford_last.rename(
    columns={"start": "W", "stop": "T", "event": "E"}
)
stanford_last_cph_wid.head()
```
![image](https://github.com/CamDavidsonPilon/lifelines/assets/20425799/c4f8dcb7-b2f1-426f-977f-5898eb81953c)

Create a `CoxPHFitter` model and fit it with the `cluster_col` specified.

```python
cph_stanford_last_wid = CoxPHFitter()
cph_stanford_last_wid.fit(
    stanford_last_cph_wid,
    duration_col="T",
    event_col="E",
    entry_col="W",
    cluster_col="id",
)
cph_stanford_last_wid.summary
```
![image](https://github.com/CamDavidsonPilon/lifelines/assets/20425799/979c5196-ee0c-4cc0-a6ac-b9ff3958d251)

However, if both a `cluster_col` and `robust` was specified, the SE value is always the same (`0.14374`) regardless of the value for `robust`.

![image](https://github.com/CamDavidsonPilon/lifelines/assets/20425799/f928da3d-ba46-4836-9147-993863f1e60b)

The standard error is different (`0.13862`) when `cluster_col` is not specified, therefore letting `robust` be set to its default value of `False`.
<img width="1029" alt="image" src="https://github.com/CamDavidsonPilon/lifelines/assets/20425799/f8ca7f67-07ed-4e10-9f13-06cd4f76541a">

lifelines version: 0.27.8
